safe/gui/analysis_utilities.py

```diff
@@ -26,151 +32,95 @@
 __revision__ = '$Format:%H$'
 
 
-def generate_impact_report(impact_function, iface):
-    """Generate the impact report from an impact function.
+def generate_report(impact_function, iface):
+    """Generate Impact Report from an Impact Function.
 
-    :param impact_function: The impact function used.
+    :param impact_function: The impact function to be used.
     :type impact_function: ImpactFunction
 
     :param iface: QGIS QGisAppInterface instance.
     :type iface: QGisAppInterface
-
     """
-    # get the extra layers that we need
+    # generate report from component definition
+    report_components = deepcopy(all_default_report_components)
+
+    # don't generate infographic if exposure is not population
+    exposure_type = layer_definition_type(impact_function.exposure)
+    if exposure_type != exposure_population:
+        report_components.remove(infographic_report)
+
+    # if exposure is population, we need to add
+    # map overview layer
+    exposure_type = layer_definition_type(impact_function.exposure)
+    if exposure_type == exposure_population:
```

Comment:
* Remove duplicated line L48 and L55.
* Use an `else` statement instead two IF doing the reverse

Reply:
upsy